### ISSUE TITLE
Using an unsupported API should raise a diagnostic

### ISSUE THREAD

**Issue by terrajobst** [dotnet/roslyn-analyzers#4013]
### Repro

```C#
using System.Runtime.Versioning;

[assembly: SupportedOSPlatform("windows")]

static class Program
{
    public static void Main()
    {
        SomeWindowsSpecific.Api();
    }
}

class SomeWindowsSpecific
{
    [UnsupportedOSPlatform("windows")]
    public static void Api() { }
}
```

### Expected

> 'Api' is unsupported by windows

### Actual

No diagnostics are raised.

**Comment by buyaa-n** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/99883854e87fce50449674b9638ece153becd082
